### PR TITLE
config: enable first fluster test for mt8195-cherry-tomato-r2

### PR DESCRIPTION
**Pull request by musamaanjum**
Enable first fluster test, AV1-TEST-VECTORS for mt8195-cherry-tomato-r2. Run the test on mainline and next until more trees are added.

Note: This is my first PR. Please review it and I'll improve it. https://github.com/kernelci/kernelci-core/pull/2535 is a pre-requisite for this.

### CODE REVIEW

**Comment by musamaanjum** [comment]
I've added the staging-skip until someone reviews to not disturb the staging.

File: config/jobs-chromeos.yaml [hunk 1]
```diff
@@ -551,3 +551,19 @@ jobs:
   tast-ui-arm64-qualcomm: *tast-ui-job
   tast-ui-x86-intel: *tast-ui-job
   tast-ui-x86-amd: *tast-ui-job
+
+  v4l2-decoder-conformance-av1:
```
Comment (laura-nao) [inline config/jobs-chromeos.yaml:555]: Given that we'll have other similar test suites where the only difference will be the `testsuite` and `decoders`  parameters, I suggest defining an anchor in this file as a base to include/extend right away - something like:
```
  v4l2-decoder-conformance-job: &v4l2-decoder-conformance-job
    template: 'v4l2-decoder-conformance-template.jinja2'
    kind: test
    params: &v4l2-decoder-conformance-job-params
      nfsroot: 'https://storage.staging.kernelci.org/rootfs/bookworm-gst-fluster/20240430.0/{debarch}/'
      job_timeout: 30
      videodec_parallel_jobs: '1'
      videodec_timeout: 90
    rules:
      tree:
        - mainline
        - next
```

The AV1 test job definition can then be simplified to:

```
  v4l2-decoder-conformance-av1:
    <<: *v4l2-decoder-conformance-job
    params:
      <<: *v4l2-decoder-conformance-job-params
      testsuite: 'AV1-TEST-VECTORS'
      decoders:
        - 'GStreamer-AV1-V4L2SL-Gst1.0'
```

Also, a gst+fluster rootfs for testing is now available here: https://storage.staging.kernelci.org/rootfs/bookworm-gst-fluster/20240430.0/ , so we can replace the nfsrootfs URL with the proper one.

File: config/jobs-chromeos.yaml [hunk 2]
```diff
@@ -551,3 +551,24 @@ jobs:
   tast-ui-arm64-qualcomm: *tast-ui-job
   tast-ui-x86-intel: *tast-ui-job
   tast-ui-x86-amd: *tast-ui-job
+
+  v4l2-decoder-conformance-job: &v4l2-decoder-conformance-job
```
Comment (laura-nao) [inline config/jobs-chromeos.yaml:555]: @musamaanjum just one more thing: the `v4l2-decoder-conformance-job` anchor should be moved higher up in this file, in the `_anchors:` section.

Comment (musamaanjum) [reply config/jobs-chromeos.yaml]: Sure.

File: config/jobs-chromeos.yaml [hunk 3]
```diff
@@ -364,6 +364,19 @@ _anchors:
         - ui.HotseatAnimation.shelf_with_navigation_widget_lacros
         - ui.WindowControl
 
+  v4l2-decoder-conformance-job: &v4l2-decoder-conformance-job
+    template: 'v4l2-decoder-conformance-template.jinja2'
+    kind: test
+    params: &v4l2-decoder-conformance-job-params
```
Comment (a-wai) [inline config/jobs-chromeos.yaml:370]:
```suggestion
    params: &v4l2-decoder-conformance-params
```

For consistency with other anchors definitions.

File: config/jobs-chromeos.yaml [hunk 4]
```diff
@@ -364,6 +364,19 @@ _anchors:
         - ui.HotseatAnimation.shelf_with_navigation_widget_lacros
         - ui.WindowControl
 
+  v4l2-decoder-conformance-job: &v4l2-decoder-conformance-job
+    template: 'v4l2-decoder-conformance-template.jinja2'
```
Comment (a-wai) [inline config/jobs-chromeos.yaml:368]:
```suggestion
    template: 'v4l2-decoder-conformance.jinja2'
```

The template shouldn't include "template" in its name, being a `.jinja2` file we already know what it is ;)

File: config/jobs-chromeos.yaml [hunk 5]
```diff
@@ -364,6 +364,19 @@ _anchors:
         - ui.HotseatAnimation.shelf_with_navigation_widget_lacros
         - ui.WindowControl
 
+  v4l2-decoder-conformance-job: &v4l2-decoder-conformance-job
```
Comment (a-wai) [inline config/jobs-chromeos.yaml:367]:
```suggestion
  v4l2-decoder-conformance: &v4l2-decoder-conformance-job
```

For consistency with other anchors definitions.

File: config/jobs-chromeos.yaml [hunk 6]
```diff
@@ -551,3 +564,11 @@ jobs:
   tast-ui-arm64-qualcomm: *tast-ui-job
   tast-ui-x86-intel: *tast-ui-job
   tast-ui-x86-amd: *tast-ui-job
+
+  v4l2-decoder-conformance-av1:
+    <<: *v4l2-decoder-conformance-job
+    params:
+      <<: *v4l2-decoder-conformance-job-params
```
Comment (a-wai) [inline config/jobs-chromeos.yaml:571]:
```suggestion
      <<: *v4l2-decoder-conformance-params
```

See suggestion above

File: config/runtime/v4l2-decoder-conformance.jinja2 [hunk 7]
```diff
@@ -0,0 +1,4 @@
+{%- set boot_commands = 'nfs' %}
+{%- set test_method = 'v4l2-decoder-conformance' %}
+{%- set base_template = 'base/' + runtime + '.jinja2' %}
+{%- extends base_template %}
```
Comment (a-wai) [inline config/runtime/v4l2-decoder-conformance.jinja2:4]: This is very similar to some of the other templates like `kselftest`... At some point we should probably make `boot_commands` and `test_method` job params and use a generic template, but that's not a requirement for this PR.

File: config/jobs-chromeos.yaml [hunk 8]
```diff
@@ -364,6 +364,19 @@ _anchors:
         - ui.HotseatAnimation.shelf_with_navigation_widget_lacros
         - ui.WindowControl
 
+  v4l2-decoder-conformance-job: &v4l2-decoder-conformance-job
+    template: 'v4l2-decoder-conformance-template.jinja2'
+    kind: test
+    params: &v4l2-decoder-conformance-job-params
+      nfsroot: 'https://storage.staging.kernelci.org/rootfs/bookworm-gst-fluster/20240430.0/{debarch}/'
+      job_timeout: 30
+      videodec_parallel_jobs: '1'
```
Comment (a-wai) [inline config/jobs-chromeos.yaml:373]:
```suggestion
      videodec_parallel_jobs: 1
```

No need for it to be a string, especially as you convert it to string in the template.

**Review by a-wai — CHANGES_REQUESTED** [review]
Some minor nits, but overall this looks quite good!

**Review by a-wai — APPROVED** [review]

**Comment by musamaanjum** [comment]
Test job link: https://lava.collabora.dev/scheduler/job/13660277
Mostly tests passed, but a few failed.

**Comment by nuclearcat** [comment]
Thanks, tested on staging, merging.

**Pull request merged**